implementations/rust/ockam/ockam_transport_tcp/src/transport/portals.rs

```diff
@@ -143,3 +146,67 @@ impl TcpTransport {
         Ok(())
     }
 }
+
+/// Result of [`TcpTransport::create_inlet`] call.
+#[derive(Clone, Debug)]
+pub struct TcpInlet {
+    socket_address: SocketAddr,
+    processor_address: Address,
+    outlet_listener_route: Arc<RwLock<Route>>,
+}
+
+impl fmt::Display for TcpInlet {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Socket: {}, Processor: {}",
+            self.socket_address, self.processor_address
+        )
+    }
+}
+
+impl TcpInlet {
+    /// Constructor
+    pub fn new(
+        socket_address: SocketAddr,
+        processor_address: Address,
+        outlet_listener_route: Arc<RwLock<Route>>,
+    ) -> Self {
+        Self {
+            socket_address,
+            processor_address,
+            outlet_listener_route,
+        }
+    }
+
+    /// Socket Address
+    pub fn socket_address(&self) -> SocketAddr {
```

Comment:
Sometimes I wish we had different names for socket "addresses" and worker "addresses".

Reply:
What do you mean? I thought that's what we have in this particular place. A dedicated type, a dedicated function name